index.js

```diff
@@ -94,8 +94,17 @@ exports.handler = async event => {
 
         // check which segments have been FULLY downloaded
         arr.segments.forEach(([firstByte, lastByte], idx) => {
-          if (arr.isLoggable(idx) && range.complete(firstByte, lastByte)) {
-            kinesisRecords.push({ ...bytesData, segment: idx })
+          if (arr.isLoggable(idx)) {
+            const rec = { ...bytesData, segment: idx }
+
+            // handle empty/zero-byte segments - just make sure their "firstByte" was downloaded
+            if (firstByte > lastByte) {
+              if (range.complete(firstByte, firstByte)) {
```

Comment:
This threw me a bit.  But imagine an arrangement:

```json
{
  "version": 4,
  "incomplete": 1
  "data": {
    "f": ["url-1.mp3", "url-2.mp3", "url-3.mp3"],
    "t": "aao",
    "a": {"f": "mp3", "b": 128, "c": 2, "s": 44100},
    "b": [1000, 2000, 2000, 3000]
  }
}
```

So `url-1.mp3` is at bytes 1000-1999, and `url-3.mp3` is 2000-3000 (the final byte is _inclusive_ in arrangements).

And `url-2.mp3` isn't really _any_ of the bytes in the file - so the existing code here will mark it as having firstByte 2000 and lastByte 1999.

I opted to wait until the byte 2000 is downloaded to log this dup-impression.  I guess really, you could log it anytime?  But this sort of made sense.